### PR TITLE
Add lint check rule for actions and handlers

### PR DESCRIPTION
**Pull request by skiddykong**

### CODE REVIEW

File: raml-lint-check/lint-check-rules/src/main/java/uk/gov/justice/services/raml/lintcheck/utils/HandlerScanner.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+public class HandlerScanner {
+
+
+    private final String basePackage;
+
+    public HandlerScanner(final String basePackage) {
+        this.basePackage = basePackage;
+    }
+
+    private Reflections configureReflections(String basePackage) {
```
Comment (mapingo) [inline raml-lint-check/lint-check-rules/src/main/java/uk/gov/justice/services/raml/lintcheck/utils/HandlerScanner.java]: basePackage needs a final

File: raml-lint-check/lint-check-rules/src/main/java/uk/gov/justice/services/raml/lintcheck/utils/RamlActionFinder.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public static Collection<String> actionsFrom(final Raml raml) {
+
+        final Collection<Resource> resources = raml.getResources().values();
+        final List<String> actionNames = new ArrayList<>();
+
+        for (final Resource resource : resources) {
+            final Collection<Action> actions = resource.getActions().values();
+            for (final Action ramlAction : actions) {
+                listOf(ramlAction.getDescription())
+                        .stream()
```
Comment (mapingo) [inline raml-lint-check/lint-check-rules/src/main/java/uk/gov/justice/services/raml/lintcheck/utils/RamlActionFinder.java]: no need for stream(), you can do forEach on a list

File: raml-lint-check/lint-check-rules/src/main/java/uk/gov/justice/services/raml/lintcheck/utils/RamlActionFinder.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+public final class RamlActionFinder {
+
+    private RamlActionFinder(){}
+
+    public static Collection<String> actionsFrom(final Raml raml) {
+
+        final Collection<Resource> resources = raml.getResources().values();
+        final List<String> actionNames = new ArrayList<>();
+
+        for (final Resource resource : resources) {
```
Comment (mapingo) [inline raml-lint-check/lint-check-rules/src/main/java/uk/gov/justice/services/raml/lintcheck/utils/RamlActionFinder.java]: You can do a forEach and use Streams for each level of this "for" statement

File: raml-lint-check/lint-check-rules/src/test/java/uk/gov/justice/services/raml/lintcheck/ActionsHaveHandlersTest.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ActionsHaveHandlersTest {
+
+    @Mock
+    LintCheckConfiguration lintCheckConfiguration;
```
Comment (mapingo) [inline raml-lint-check/lint-check-rules/src/test/java/uk/gov/justice/services/raml/lintcheck/ActionsHaveHandlersTest.java]: can be private

File: raml-lint-check/lint-check-rules/src/test/java/uk/gov/justice/services/raml/lintcheck/it/ActionsHaveHandlersMojoTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        assertTrue( pom.exists() );
+
+        final LintCheckMojo myMojo = (LintCheckMojo) lookupConfiguredMojo(pom, "lint-check");
+
+        assertNotNull( myMojo );
+
+        try {
+            myMojo.execute();
+            fail();
+        } catch (MojoExecutionException e) {
```
Comment (mapingo) [inline raml-lint-check/lint-check-rules/src/test/java/uk/gov/justice/services/raml/lintcheck/it/ActionsHaveHandlersMojoTest.java]: Could use the JUnit
    @Rule
    public ExpectedException expectedException = ExpectedException.none();

Comment (skiddykong) [inline raml-lint-check/lint-check-rules/src/test/java/uk/gov/justice/services/raml/lintcheck/it/ActionsHaveHandlersMojoTest.java]: gave this a try based on the examples in spec: http://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html

Was not able to get it to work. It also does not give me an easy way to analyze the contents of the causing exception message.

**Pull request merged**